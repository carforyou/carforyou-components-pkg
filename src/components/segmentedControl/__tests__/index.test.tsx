import React from "react"
import { render, fireEvent } from "@testing-library/react"

import SegmentedControl from "../index"

describe("<SegmentedControl>", () => {
  const mockedOnSelect = jest.fn()

  const options = [
    { value: 1, name: "One" },
    { value: 2, name: "Two" },
    { value: 3, name: "Three" }
  ]

  const renderWrapper = (data = {}) => {
    const { renderOption } = { renderOption: undefined, ...data }
    return render(
      <SegmentedControl
        options={options}
        onSelect={mockedOnSelect}
        renderOption={renderOption}
      />
    )
  }

  beforeEach(mockedOnSelect.mockClear)

  it("passes selection to callback", () => {
    const { getByText } = renderWrapper()

    const button = getByText("Two")
    fireEvent.click(button)

    expect(mockedOnSelect).toHaveBeenCalledWith(2)
  })

  it("highlights newly selected element", () => {
    const { getByText } = renderWrapper()

    const button = getByText("Two")
    fireEvent.click(button)

    expect(getByText("Two").classList).toContain("bg-teal")
  })

  it("does not select already selected element", () => {
    const { getByText } = renderWrapper()

    const button = getByText("One")
    fireEvent.click(button)

    expect(mockedOnSelect).not.toHaveBeenCalled()
  })

  it("accepts custom render for options", () => {
    const { getByText } = renderWrapper({
      renderOption: ({ name, value }) => (
        <div>
          {name} ({value})
        </div>
      )
    })

    options.forEach(({ value, name }) =>
      expect(getByText(`${name} (${value})`))
    )
  })

  it("has correct padding without the custom renderer", () => {
    const { getByText } = renderWrapper()
    const button = getByText("Two")

    expect(button.classList).toContain("py-16")
    expect(button.classList).toContain("px-8")
  })

  it("passes correct padding to the custom renderer", () => {
    const { getByText } = renderWrapper({
      renderOption: ({ name }) => <div>{name}</div>
    })

    const option = getByText("Two")
    expect(option.classList).toContain("py-16")
    expect(option.classList).toContain("px-8")
  })
})
