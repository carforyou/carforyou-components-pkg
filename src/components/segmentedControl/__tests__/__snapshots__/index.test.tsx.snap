// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SegmentedControl> renders a div tag for a custom renderer with link 1`] = `
<div
  class="flex justify-center items-center leading-xs transition duration-200 cursor-pointer font-bold text-base border-t-2 border-b-2 border-r-2 focus:outline-none bg-transparent text-teal w-12/12 border-teal focus:shadow-focus rounded-none hover:bg-teal-light active:bg-teal-bright"
>
  <a
    class="w-12/12 hover:opacity-100 flex flex-col"
  >
    <span
      class="flex items-center justify-center px-8 py-16"
    >
      Two
    </span>
  </a>
</div>
`;
